Bottom: adf6e1db12a2c0d2d284dda25aa6bcd859db7073
Top:    590d78abca6568895476bf489103bd40362432d6
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 13:37:16 +0800

autofs-5.0.7 - add mapent null check in lookup_nisplus.c:lookup_mount()

malloc(3) could return null under low memory conditions, add a null check
for this case.


---

diff --git a/modules/lookup_nisplus.c b/modules/lookup_nisplus.c
index 8237a1e..ef942a7 100644
--- a/modules/lookup_nisplus.c
+++ b/modules/lookup_nisplus.c
@@ -584,7 +584,8 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		if (me && (me->source == source || *me->key == '/')) {
 			mapent_len = strlen(me->mapent);
 			mapent = malloc(mapent_len + 1);
-			strcpy(mapent, me->mapent);
+			if (mapent)
+				strcpy(mapent, me->mapent);
 		}
 	}
 	cache_unlock(mc);
